WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -660,14 +661,18 @@ private Script GetCoordinatorScriptPreventReuse(Round round)
 		return coordinatorScriptPubKey;
 	}
 
-	private void CoinVerifier_CoinBlacklisted(object? _, Coin coin)
+	private async void CoinVerifier_CoinBlacklistedAsync(object? _, Coin coin)
 	{
-		// For logging reason Prison needs the roundId.
-		var roundState = RoundStates.FirstOrDefault(rs => rs.CoinjoinState.Inputs.Any(input => input.Outpoint == coin.Outpoint));
+		using CancellationTokenSource cts = new(TimeSpan.FromMinutes(1));
+		using (await AsyncLock.LockAsync(cts.Token).ConfigureAwait(false))
+		{
+			// For logging reason Prison needs the roundId.
+			var round = Rounds.FirstOrDefault(round => round.Alices.Select(alice => alice.Coin.Outpoint).Contains(coin.Outpoint));
```

Comment:
Why does `Rounds` include the coin but `RoundStates` not?

Reply:
So, I assume what happens in `master` is the following:

We were checking `RoundStates.FirstOrDefault(rs => rs.CoinjoinState.Inputs....)`, but we are in the end of input registration (This is a `CoinVerifier` event handler), so the inputs are not "commited"/finalized.

By not commited, I mean the `RoundState.CoinjoinState`, which is a `MultipartyTransactionState`, is not yet updated with the inputs. (because the conn confirmation phase didn't happen yet? I'm not sure)

Reply:
Oh yes, you are right!

Reply:
Yes, this is what is happening.